### PR TITLE
web(feat): Add support for actions when clicking notifications

### PR DESCRIPTION
**Pull request by bo0tzz**
This PR adds a `NotificationAction` to be taken when a notification popup card is clicked. The two currently implemented actions are discard (remove the card, default) and link (open a link in a new tab). The error notification that is displayed when trying to upload more than 50 assets at once now links to the CLI guide when clicked.

There's probably a nicer way to structure the `NotificationAction` type, but I don't know my way around typescript well enough to come up with it. Any suggestions for improvements there are welcome. If not, this PR is ready to merge from my end :)

### CODE REVIEW

File: web/src/lib/components/shared-components/notification/notification.ts
Comment (bo0tzz) [inline web/src/lib/components/shared-components/notification/notification.ts:18]: There's probably a better way to do this?

Comment (bo0tzz) [inline web/src/lib/components/shared-components/notification/notification.ts:18]: There's probably a better way to do this?

**Comment by jrasm91** [comment]
This is great, except having to explain the notification is clickable. Why about allow something like this instead?

image

Allow html in messages
```
<p class="text-sm pl-[28px] pr-[16px]" data-testid="message">{@html notificationInfo.message}</p>
```

Pass link in the message
```
`To upload more than 50 files at a time, please see <a href="https://immich.app/docs/usage/bulk-upload" class="underline">Bulk Upload</a> in the docs.`
```

**Comment by bo0tzz** [comment]
I tried thinking of other ways to indicate the notification being clickable but couldn't come up with something good. I hadn't realised it's that easy to accept html in the message though. That's a good idea, I'll add that. Thanks!

**Comment by jrasm91** [comment]
I think the default and only behavior for clicking the card should be to dismiss it. I don't think that even needs to be something that would add to the notification interface, that's just how clicking notifications work. Then you could remove the prevent default as well as the window.open(action.target), just leave that to the embedded link in the message.

**Comment by jrasm91** [comment]
I think adding additional notification click behavior is a good idea, but for your specific use case we don't need it. I'd say either (1) remove the action from the notification in `file-upload` or (2) leave text-underline around `Bulk Import`, but don't make it an actual link, since the whole card is basically a link.

**Comment by bo0tzz** [comment]
@alextran1502 what are your thoughts on having the whole card be a clickable link vs. just having a link in the message (and card-click always being discard)?

**Comment by bo0tzz** [comment]
> It would be nice if the notification was flexible  

That was the idea behind the current implementation, with 2 and 4 currently implemented, and room for adding the others later when we need them.

**Comment by jrasm91** [comment]
Is it just a matter of not using a <a> at the same time as the link action? Like using one or the other, but not both?

**Comment by bo0tzz** [comment]
Yeah, not using both at the same time would remove the need for that `preventDefault`. In that case, when using `<a>`, it might be good to have a no-op Action so a misclick doesn't remove the whole card while you're trying to open the link?

**Comment by jrasm91** [comment]
Yeah, that's a good idea.

File: web/src/lib/components/shared-components/notification/notification.ts
Comment (alextran1502) [inline web/src/lib/components/shared-components/notification/notification.ts:12]: Maybe we can use the property, `dismissible?: boolean` to indicate that the notification card can be dimissed by the user? Then in the `notification-card.svelte` you can use the `{#if} {/if}` syntax to show the dimiss button/icon

Comment (bo0tzz) [reply web/src/lib/components/shared-components/notification/notification.ts:12]: Are there any notifications that we would not want to be dismissible?

File: web/src/lib/components/shared-components/notification/notification-card.svelte
Comment (alextran1502) [inline web/src/lib/components/shared-components/notification/notification-card.svelte:93]: with the `@html` tag, it should render the string with valid html syntax, right?

File: web/src/lib/components/shared-components/notification/notification.ts
Comment (alextran1502) [reply web/src/lib/components/shared-components/notification/notification.ts:12]: I think most of the info should not be dismissible, any notification card that might pop up constantly should have this flag enable, like error notification which might have multiple popup in a row

**Comment by alextran1502** [comment]
> @alextran1502 what are your thoughts on having the whole card be a clickable link vs. just having a link in the message (and card-click always being discard)?

I don't have strong feelings here, either way works for me!

File: web/src/lib/components/shared-components/notification/notification-card.svelte
Comment (bo0tzz) [reply web/src/lib/components/shared-components/notification/notification-card.svelte:93]: It properly renders html tags, yes.

File: web/src/lib/components/shared-components/notification/notification.ts
Comment (bo0tzz) [reply web/src/lib/components/shared-components/notification/notification.ts:12]: I would argue for notifications being dismissible by default instead. Especially on the main gallery view, I can imagine wanting to get rid of a card so you can click something that's underneath. Especially if there are multiple (of any type), waiting for it to disappear could get annoying.

Comment (alextran1502) [reply web/src/lib/components/shared-components/notification/notification.ts:12]: That works too, so we have the `dismissible` property is true by default, is that how you are thinking?

Comment (alextran1502) [reply web/src/lib/components/shared-components/notification/notification.ts:12]: Fair point. We can add that in if there is one that we don't want to be dismissed later on. You can merge this PR if you are done with it

**Review by alextran1502 — APPROVED** [review]

**Comment by bo0tzz** [comment]
I've now removed the `<a>` element from the file upload count error and replaced it with just an underline, with the whole card being a link to the docs. If it's preferred to swap that around (link in text and card click being discard), I can do that. Otherwise, this is ready to merge IMO :)

**Pull request merged**
